### DISCUSSION TITLE
Enhancement: Chat Folders and subfolders

### DISCUSSION THREAD

**Discussion by MontDawgg** [danny-avila/LibreChat#410] [Feature Requests & Suggestions]
### Contact Details

montediaz@live.com

### What features would you like to see added?

**Description:**
It would be an amazing addition if we could organize our chats into folders and subfolders. Having up to 3 subfolders is enough to cover 99% of the userbase.

**Proposed Solution:**
Let us create folders to organize our chats. The ability to color code them would be cool but not a necessity for this functionality.  Additionally, a "recycle bin" or "trash" folder would be helpful. 

### More details

Many users have hundreds of chats, and it becomes necessary to organize them efficiently to improve workflow. In other apps where this functionality has been implemented it has been an extremely popular option.

### Pictures

![240754284-cf983871-c494-417f-be69-341e6f41f72e](https://github.com/danny-avila/chatgpt-clone/assets/133587103/85e6df6d-23a0-45f0-8ee4-978568e6d2a7)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Yes I agree and this is planned for me to implement

**Comment by secondtruth** [comment]
This is a great idea, especially with the colors. I hope this feature will be implemented.

**Comment by iamNoah1** [comment]
@danny-avila whats the status of the folder feature?

**Comment by raymelon** [comment]
Hey @danny-avila! I am curious if this feature going to be released soon? I just wanted to know if it is in the roadmap. No worries if it's not.

**Reply by danny-avila** [reply]
It is already "released" via "Bookmarks": https://www.youtube.com/watch?v=cvosUxogdpI

**Reply by raymelon** [reply]
Oh, that's cool @danny-avila! Sorry I missed that.

Maybe because I was expecting a straightforward user flow just like in Chatbot UI, where you can simply create a folder and put drag any conversation you like to that folder, all in the left sidebar where other conversations reside.

Would be very cool to have an alternative sidebar like this, aside from the default sidebar that shows every chat in reverse chronological order.

For reference,

![image](https://github.com/user-attachments/assets/e2583fca-488f-49e7-980b-bb52b51008ee)

A simpler implementation would be an ability to pin this to the sidebar, instead of always launching it from a drawer

![image](https://github.com/user-attachments/assets/88f511c5-ed8f-4c0c-825d-cca971f2d60c)

Got the idea from this (from the original poster),

![image](https://github.com/user-attachments/assets/cf4e4728-47a8-4019-a2a1-258aa1d01998)
